tests/make/make.spec.js

```diff
@@ -2,207 +2,157 @@ const { spawnSync } = require('child_process');
+    'https://github.com/mozilla/addons-server',
   );
+});
 
-  test('docker compose substitution', () => {
-    expect(new Set(Object.keys(defaultValues))).toStrictEqual(new Set(keys));
+test('map docker compose config', () => {
+  const values = {
+    DOCKER_VERSION: 'version',
+    HOST_UID: 'uid',
+    SUPERUSER_EMAIL: 'email',
+    SUPERUSER_USERNAME: 'name',
+  };
 
-    const composeFiles = globSync('docker-compose*.yml', { cwd: rootPath });
-    const variableDefinitions = [];
+  runSetup(values);
 
-    for (let file of composeFiles) {
-      const fileContent = fs.readFileSync(path.join(rootPath, file), {
-        encoding: 'utf-8',
-      });
+  const { stdout: rawConfig } = spawnSync(
+    'docker',
+    ['compose', 'config', 'web', '--format', 'json'],
+    { encoding: 'utf-8' },
+  );
 
-      for (let line of fileContent.split('\n')) {
-        const regex = /\${(.*?)(?::-.*)?}/g;
-        let match;
-        while ((match = regex.exec(line)) !== null) {
-          const variable = match[1];
-          variableDefinitions.push(variable);
-        }
-      }
-    }
+  const config = JSON.parse(rawConfig);
+  const { web } = config.services;
 
-    for (let variable of variableDefinitions) {
-      expect(keys).toContain(variable);
-    }
+  expect(web.image).toStrictEqual(
+    `mozilla/addons-server:${values.DOCKER_VERSION}`,
+  );
+  expect(web.platform).toStrictEqual('linux/amd64');
+  expect(web.environment.HOST_UID).toStrictEqual(values.HOST_UID);
+  expect(web.environment.SUPERUSER_EMAIL).toStrictEqual(values.SUPERUSER_EMAIL);
+  expect(web.environment.SUPERUSER_USERNAME).toStrictEqual(
+    values.SUPERUSER_USERNAME,
+  );
+  expect(config.volumes.data_mysqld.name).toStrictEqual(
+    'addons-server_data_mysqld',
+  );
+});
+
+function gitConfigUserEmail() {
+  const { stdout: value } = spawnSync('git', ['config', 'user.email'], {
+    encoding: 'utf-8',
   });
 
-  describe.each(product(configurableKeys, options, options, options))(
-    'test_configurable_keys',
-    (name, useFile, useEnv, useArgs) => {
-      it(`test_${name}_file:${useFile}_env:${useEnv}_args:${useArgs}`, () => {
-        let expectedValue = defaultValues[name];
+  return value.trim() || 'admin@mozilla.com';
+}
 
-        if (!expectedValue) throw new Error(`expected value for ${name}.`);
+function gitConfigUserName() {
+  const { stdout: value } = spawnSync('git', ['config', 'user.name'], {
+    encoding: 'utf-8',
+  });
+  return value.trim() || 'admin';
+}
+
+function standardPermutations(name, defaultValue) {
+  return [
+    {
+      name,
+      file: undefined,
+      env: undefined,
+      expected: defaultValue,
+    },
+    {
+      name,
+      file: 'file',
+      env: undefined,
+      expected: 'file',
+    },
+    {
+      name,
+      file: undefined,
+      env: 'env',
+      expected: 'env',
+    },
+    {
+      name,
+      file: 'file',
+      env: 'env',
+      expected: 'env',
+    },
+  ];
+}
+
+const testCases = [
```

Comment:
This format for the env variable test is much more flexible and simple. Also results in less code.